From: Robert Schwebel <r.schwebel@pengutronix.de
Subject: fix long double handling

On x86 32 bit, we get this error:

src/x86/ffi.c: In function 'ffi_prep_cif_machdep':
src/x86/ffi.c:137: error: duplicate case value
src/x86/ffi.c:136: error: previously used here
make[3]: *** [src/x86/ffi.lo] Error 1

due to the fact that we don't HAVE_LONG_DOUBLE. Fix it.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---
 src/x86/ffi.c |    3 +++
 1 file changed, 3 insertions(+)

Index: libffi-3.0.7/src/x86/ffi.c
===================================================================
--- libffi-3.0.7.orig/src/x86/ffi.c
+++ libffi-3.0.7/src/x86/ffi.c
@@ -31,6 +31,7 @@
 #ifndef __x86_64__
 
 #include <ffi.h>
+#include <fficonfig.h>
 #include <ffi_common.h>
 
 #include <stdlib.h>
@@ -134,7 +135,9 @@ ffi_status ffi_prep_cif_machdep(ffi_cif 
     case FFI_TYPE_SINT64:
     case FFI_TYPE_FLOAT:
     case FFI_TYPE_DOUBLE:
+#ifdef HAVE_LONG_DOUBLE
     case FFI_TYPE_LONGDOUBLE:
+#endif
       cif->flags = (unsigned) cif->rtype->type;
       break;
 
